pkg/capabilities/capabilities_test.go

```diff
@@ -0,0 +1,61 @@
+package capabilities
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/syndtr/gocapability/capability"
+)
+
+type fakeCapability struct {
+	capability.Capabilities
+
+	get  func(capability.CapType, capability.Cap) bool
+	load func() error
+}
+
+func (f fakeCapability) Get(which capability.CapType, what capability.Cap) bool {
+	if f.get != nil {
+		return f.get(which, what)
+	}
+	return true
+}
+
+func (f fakeCapability) Load() error {
+	if f.load != nil {
+		return f.load()
+	}
+	return nil
+}
+func TestCheckRequiredCapabilities(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		require.NoError(t, CheckRequiredCapabilities(fakeCapability{}, []capability.Cap{capability.CAP_SYS_ADMIN, capability.CAP_IPC_LOCK}))
+	})
+
+	t.Run("missing CAP_SYS_ADMIN", func(t *testing.T) {
+		err := CheckRequiredCapabilities(fakeCapability{get: func(capType capability.CapType, c capability.Cap) bool {
+			assert.Equal(t, capability.EFFECTIVE, capType)
+			assert.Equal(t, capability.CAP_SYS_ADMIN, c)
+			return false
+		}}, []capability.Cap{capability.CAP_SYS_ADMIN})
+		assert.Equal(t, "insufficient privileges to run: missing CAP_SYS_ADMIN", err.Error())
+	})
+}
+
+func TestLoadSelfCapabilities(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
```

Comment:
Do you think we can really unit test this method? It seems that we're actually testing the implementation of the fakeCapability. If you comment out the line `err := selfCap.Load()` in the `LoadSelfCapabilities` method the test called "happy path" is still green.

BTW, what happy path means in this context?

Reply:
Yeah that's a good point - I think another assertion that's missing is to assert if Load() was actually called or not. What do you think about that? Personally, I'm not fully satisfied with it either. Any ideas?

Happy path in this context would mean if the following items take place:
1. Load() is called and returns without errors.
2. Load() is called the required number of times that it's supposed to be (in this case once).

Reply:
Please see the idea mentioned above demonstrated here https://github.com/aquasecurity/tracee/pull/1256/commits/60d0a30163a9e5258d4bb427a49a65c61b1462c3